src/main/java/uk/gov/homeoffice/digital/sas/balancecalculator/module/NightHoursAccrualModule.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.homeoffice.digital.sas.balancecalculator.module;
+
+import static uk.gov.homeoffice.digital.sas.balancecalculator.utils.RangeUtils.UK_TIME_ZONE;
+
+import com.google.common.collect.Range;
+import com.google.common.collect.RangeSet;
+import com.google.common.collect.TreeRangeSet;
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.accrual.enums.AccrualType;
+
+public class NightHoursAccrualModule extends AccrualModule {
+
+  public NightHoursAccrualModule() {
+    super(AccrualType.NIGHT_HOURS);
+  }
+
+  @Override
+  public BigDecimal calculateShiftContribution(ZonedDateTime startTime, ZonedDateTime endTime) {
+    ZonedDateTime ukStartTime = startTime.withZoneSameInstant(UK_TIME_ZONE);
+    ZonedDateTime ukEndTime = endTime.withZoneSameInstant(UK_TIME_ZONE);
+
+    RangeSet<ZonedDateTime> timeEntryNightHours =
+        getIntersectionWithNightHours(ukStartTime, ukEndTime);
+
+    return timeEntryNightHours.asRanges().stream()
+        .map(range -> new BigDecimal(
+            Duration.between(range.lowerEndpoint(), range.upperEndpoint()).toMinutes()))
+        .reduce(BigDecimal.ZERO, BigDecimal::add);
+  }
+
+  private RangeSet<ZonedDateTime> getIntersectionWithNightHours(ZonedDateTime startTime,
+                                                                ZonedDateTime endTime) {
+
+    ZonedDateTime midnightCurrentDay = atTime(startTime, 0);
+    ZonedDateTime endOfPreviousDayNightHours = atTime(startTime, 6);
```

Comment:
Would it be useful to have these values as constants? Thinking just for prominence if nothing else.